src/include/miopen/hipoc_kernel.hpp

```diff
@@ -47,36 +48,20 @@ inline HipEventPtr make_hip_event()
 
 #if 1 // Keep around other storage techinques -- @pfultz2 27.03.2017
 
-#if 0 // Keep around other storage techinques -- @pfultz2 27.03.2017
-template <class T, class U>
-struct KernelArgsPair
-{
-    static const int alignment    = sizeof(U);
-    static const int padding      = (alignment - sizeof(T) % alignment) % alignment;
-    static const int second_index = sizeof(T) + padding;
-    KernelArgsPair(T x, U y)
-    {
-        new(buffer) T(x); // NOLINT (clang-analyzer-cplusplus.PlacementNew)
-        new(buffer + second_index) U(y);
-    }
-    alignas(U) char buffer[second_index + sizeof(U)] = {};
-};
-#else
 template <class T, class U>
 struct KernelArgsPair
 {
-    static const int alignment = alignof(U);
-    static const int padding   = (alignment - (sizeof(T) % alignment)) % alignment;
-    static_assert(padding >= 0, "padding cannot be negative");
-    static const int second_index = sizeof(T) + padding;
+    constexpr static const auto alignU       = alignof(U);
+    constexpr static const auto padding      = (alignU - (sizeof(T) % alignU)) % alignU;
+    constexpr static const auto second_index = sizeof(T) + padding;
```

Comment:
- Please remove `const`: every `constexpr` is implicitly `const` so .
- Please remove `static`: `static` prevents objects from being optimized out.

Reply:
Will remove `const` in favor of `constexpr` but class data members must be `static` to be able to use `constexpr` declaration.

Reply:
@amberhassaan is it so that you get the build error if static is removed?

Reply:
Exactly https://godbolt.org/z/Wo8j3no3j

Reply:
@amberhassaan [resolved]